pennylane/templates/subroutines/select_pauli_rot.py

```diff
@@ -90,7 +90,7 @@ def __init__(
         self.hyperparameters["target_wire"] = qml.wires.Wires(target_wire)
         self.hyperparameters["rot_axis"] = rot_axis
 
-        if qml.math.shape(angles)[0] != 2 ** len(control_wires):
+        if qml.math.shape(angles)[-1] != 2 ** len(control_wires):
```

Comment:
This was preventing broadcasting of `SelectPauliRot`, although its methods all seem to be fine with broadcasting :D